addon/-private/functional/modifier-manager.ts

```diff
@@ -28,15 +29,25 @@ function setup(
   MODIFIER_TEARDOWNS.set(modifier, teardown);
 }
 
+type CreateModifierArgs313 = [Factory];
+type CreateModifierArgs322 = [FunctionalModifier];
+type CreateModifierArgs = CreateModifierArgs313 | CreateModifierArgs322;
+
 class FunctionalModifierManager {
-  capabilities = capabilities('3.13');
+  capabilities = capabilities(gte('3.22.0') ? '3.22' : '3.13');
 
-  createModifier(factory: Factory): FunctionalModifier {
+  createModifier(...[factoryOrClass]: CreateModifierArgs): FunctionalModifier {
     // This looks superfluous, but this is creating a new instance
     // of a function -- this is important so that each instance of the
     // created modifier can have its own state which is stored in
     // the MODIFIER_ELEMENTS and MODIFIER_TEARDOWNS WeakMaps
-    return (...args) => factory.class(...args);
+    return (...args) => {
+      if ('class' in factoryOrClass) {
```

Comment:
should this also be wrapped in a `if ember@3.13`? do the comparability helpers get burned in at build time? (like DEBUG?)

Reply:
Yes, this should also be guarded (to only be done when we know it needs to be).